mage_integrations/mage_integrations/sources/api/__init__.py

```diff
@@ -138,51 +184,82 @@ def load_data(
         response_parser = self.config.get('response_parser')
         columns = self.config.get('columns')
 
-        response = self.__build_response()
-        result = response.json()
+        separator = self.config.get('separator')
+        if separator is None:
+            separator = ','
 
-        if response_parser:
-            result = dig(result, response_parser)
+        header = self.config.get('has_header')
+        if header is None:
+            header = False
 
-        sample = None
-        requires_columns = False
-        if result and len(result) >= 1:
-            sample = result[0]
-            requires_columns = type(sample) is not dict
+        response = self._build_response()
 
-        rows = []
+        checked_type = self._check_response_type(response)
 
-        if requires_columns:
-            if not columns or len(columns) == 0:
-                col_length = 0
-                if type(sample) is list:
-                    max_length = 0
+        if checked_type == 'text/plain':
+            df = polars.read_csv(StringIO(response.content.decode()), separator=separator,
+                                 has_header=header).to_pandas()
+            yield df
```

Comment:
It should yield a list of dictionary. You can do `yield df.to_dicts()`. Same below.

Reply:
fixed on 54116c86e7bda840fbc5c3ef46dd8bfe5ea0ce7d